src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/MethodBuilder.Mono.cs

```diff
@@ -568,8 +568,6 @@ public override MethodInfo MakeGenericMethod(params Type[] typeArguments)
                 throw new InvalidOperationException("Method is not a generic method definition");
             if (typeArguments == null)
                 throw new ArgumentNullException(nameof(typeArguments));
-            if (generic_params!.Length != typeArguments.Length)
```

Comment:
I'm not sure I understood why this is wrong? It sounds like they should be the same; what do they really represent?

Reply:
A generic method is something like this `void Swap<T>(T x, T y)`. `generic_params` holds `T x, T y`. `typeArguments` holds a type that you want T to be. The lengths of them are irrelevant to each other.

Additionally, I checked the implementation on CoreCLR side, this check doesn't exist there. Here are the CoreCLR implementations:
- On one side, `MethodBuilder::MakeGenericMethod` https://github.com/dotnet/runtime/blob/0fd78a827e0da866174b5a70fd45515da9b7c0a7/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/MethodBuilder.cs#L543-L546 calls https://github.com/dotnet/runtime/blob/master/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/MethodBuilderInstantiation.cs#L11-L17 which eventually calls the constructor
- On the other side, the constructor of `MethodOnTypeBuilderInstantiation` assign type to a variable of `TypeBuilderInstantiation` type. And `MakeGenericMethod ` of `TypeBuilderInstantiation` could be found here https://github.com/dotnet/runtime/blob/1cdafd27e4afd2c916af5df949c13f8b373c4335/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilderInstantiation.cs#L20-L37

Reply:
@naricc Evidently CoreCLR actually ends up generating some invalid IL (ie IL that applies the generic method to an incorrect number of arguments) and then only throws an error at runtime  https://gist.github.com/lambdageek/e8173ed7b0e3f416ae1ff656d5141f98

Reply:
@lambdageek, if this is a potential bug in coreclr implementation, should mono introduce the same bug or report it instead to find resolution?

Reply:
@am11 I don't think it's a bug - or at least I could conceive of a rationalization for why CoreCLR behaves as it does.  Just because IL _today_ does not allow - for example - defaulted generic parameters, doesn't mean that it won't support it in the future.  So while the IL is bad today, you could imagine using that at some point in the future it might actually be sensible in some circumstances. So it makes sense to allow the IL to be constructed and to leave the check is at the place where it actually matters - in the runtime when it needs to decide how to execute the IL.

(By defaulted generic parameters I mean something like ``` void Method<T1, T2> (T1 arg1, T2 arg2) where T2 : default(T2)=int``` and then uses could be `Method<string>("abcd", 42) // actually Method<string,int>(...)` )